shim/shim.c

```diff
@@ -432,6 +446,7 @@ handle_proxy_output(struct cc_shim *shim)
 		code = *(buf + STREAM_HEADER_SIZE); 	// hyperstart has sent the exit status
 		shim_debug("Exit status for container: %d\n", code);
 		free(buf);
+		restoreTerminal();
```

Comment:
It might be safer to also call this when receiving certain signals to ensure the user is returned to a sane terminal environment even if the shim crashes.